core/src/main/scala/org/apache/spark/executor/MesosExecutorBackend.scala

```diff
@@ -81,6 +88,16 @@ private[spark] class MesosExecutorBackend
       env)
   }
 
+  private def getRangesFromResources(executorInfo: ExecutorInfo): List[(Long, Long)] = {
+  val ranges = executorInfo.getResourcesList.asScala
+    .filter(r => r.getType == Value.Type.RANGES & r.getName == "ports")
+    .toList.map(r => r.getRanges.getRangeList)
+
+    // a final list of all ranges in format (Long , Long)
+    ranges.flatMap { ranges => ranges.asScala.map { r => (r.getBegin, r.getEnd) }}
```

Comment:
scalastyle is incomplete. I think it gets us 60% of the way there but there are still many things it doesn't catch.